Zend/zend_compile.h

```diff
@@ -310,6 +310,13 @@ typedef struct _zend_property_info {
 #define OBJ_PROP_TO_NUM(offset) \
 	((offset - OBJ_PROP_TO_OFFSET(0)) / sizeof(zval))
 
+typedef struct _zend_class_constant {
+	zval value;
+	uint32_t flags;
```

Comment:
good idea. I also though about this, but then forgot :)
I'll try to change this tomorrow.

On Mon, Dec 7, 2015 at 7:14 PM, Xinchen Hui <notifications@github.com>
wrote:

> In Zend/zend_compile.h
> <https://github.com/php/php-src/pull/1662#discussion_r46839438>:
>
> > @@ -310,6 +310,13 @@ typedef struct _zend_property_info {
> >  #define OBJ_PROP_TO_NUM(offset) \
> >  	((offset - OBJ_PROP_TO_OFFSET(0)) / sizeof(zval))
> >
> > +typedef struct _zend_class_constant {
> > +	zval value;
> > +	uint32_t flags;
>
> maybe you can use zval.u2.var_flags instead of flags here? which should
> save a pointer size for this struct
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1662/files#r46839438>.
>